nexus/src/app/deployment.rs

```diff
@@ -151,6 +156,47 @@ impl super::Nexus {
             NEXUS_REDUNDANCY,
         )?;
 
+        let service_external_ips = datastore
+            .service_ip_all_list_batched(opctx)
+            .await?
+            .into_iter()
+            .filter_map(|external_ip| {
+                if !external_ip.is_service {
```

Comment:
It would definitely be a major bug in the datastore lookup, yes. We could omit this check entirely. The query for nics just below this doesn't have an equivalent check because the type we get back from the query doesn't even give us an opportunity to check it; arguably the datastore could also use more refined types for external IPs, too? I'm ambivalent about that, and almost left this check out in the first place.

Reply:
I left this in because our current `ExternalIp` type doesn't really go for "make illegal states unrepresentable"; e.g., if `is_service` is true, then `parent_id` should always be `Some(_)`, but I feel like since that illegal state _is_ representable, it's safer to check here before converting to a type that does not allow that illegal state.

Reply:
Leave it in, make it an error?

Reply:
Now an `error!` [35b1e16](https://github.com/oxidecomputer/omicron/pull/5344/commits/35b1e16bbcc0e9feacca4242b8d344dbe57169d9)